normandy/recipes/filters.py

```diff
@@ -157,6 +157,41 @@ def capabilities(self):
         return set()
 
 
+class PlatformFilter(BaseFilter):
+    """Match a user based on what operating system they are using.
+
+    .. attribute:: type
+
+        ``platform``
+
+    .. attribute:: platforms
+
+        List of platforms to filter against. The choices are `all_linux`,
+        `all_windows`, and `all_mac`.
+
+        :example: ``["all_windows", "all_linux"]``
+    """
+
+    type = "platform"
+    platforms = serializers.ListField(child=serializers.CharField(), min_length=1)
+
+    def to_jexl(self):
+        platforms_jexl = []
+        for platform in self.initial_data["platforms"]:
+            if platform == "all_mac":
+                platforms_jexl.append("normandy.os.isMac")
+            elif platform == "all_windows":
+                platforms_jexl.append("normandy.os.isWindows")
+            elif platform == "all_linux":
+                platforms_jexl.append("normandy.os.isLinux")
```

Comment:
Shouldn't there be an error if there is an unrecognized platform in the list?

Reply:
I thought about that... but I also thought that this was probably going to be coming in from a multiselect... I thought the possibility of something else getting entered was slim. I can add it though.

Reply:
One of the rules for Normandy server is to always double check the client. We cannot assume that this is coming from a multi-select, or even from a client that we control. In the worst case, this could be coming from an attacker that has slipped through other defenses. In the more realistic case, the data could be coming from a mythmon who is using a CLI because I couldn't the normal UI to do something. (I actually did this today to debug a problem).

Generally, we should consider what this API looks like independent of an UI. In fact, this API *is* our interface. The user of the interface are tools like NDT, Edi, Firefox, and the Data Pipeline. Within this hierarchy, NDT doesn't get special privilege. It's just another user.

Reply:
OK I understand and will add it to this one, as well as future ones. Thanks.